### PR TITLE
Fix stable branch to build on Rubies 1.8.7~2.4.0

### PR DESCRIPTION
**Pull request by zzak**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,26 +6,20 @@ rvm:
...
   - 2.0.0
-  - 2.1
-  - 2.2
-  - 2.3.0
+  - 2.1.10
+  - 2.2.6
+  - 2.3.3
+  - 2.4.0
   - rbx-2
   - jruby
```
Comment (mwpastore) [inline .travis.yml]: Should we pin this to 9.0.5.0?

File: .travis.yml [hunk 2]
```diff
@@ -6,26 +6,20 @@ rvm:
...
   - 1.9.3
   - 2.0.0
-  - 2.1
-  - 2.2
-  - 2.3.0
+  - 2.1.10
+  - 2.2.6
+  - 2.3.3
+  - 2.4.0
   - rbx-2
```
Comment (mwpastore) [inline .travis.yml]: I feel like we should either fix Rubinius, or remove it from Travis altogether.

File: Gemfile [hunk 3]
```diff
@@ -10,30 +10,19 @@ RUBY_ENGINE = 'ruby' unless defined? RUBY_ENGINE
 source 'https://rubygems.org' unless ENV['QUICK']
 gemspec
 
-gem 'rake'
+gem 'rake', '~> 10.0'
 gem 'rack-test', '>= 0.6.2'
 gem "minitest", "~> 5.0"
 
-# Allows stuff like `tilt=1.2.2 bundle install` or `tilt=master ...`.
```
Comment (mwpastore) [inline Gemfile]: My only concern with removing this is that someone might be using it in production.

File: Gemfile [hunk 4]
```diff
@@ -44,7 +33,7 @@ if RUBY_ENGINE == "ruby" and RUBY_VERSION > '1.9.2'
   gem 'puma'
   gem 'net-http-server'
   gem 'yajl-ruby'
-  gem 'nokogiri'
+  gem 'nokogiri', '~> 1.5'
```
Comment (jkowens) [inline Gemfile]: Try pinning to nokogiri 1.6.8.1. Versions after that require Ruby 2.1.0 or greater.

File: .travis.yml (hunk 1 above)
Comment (zzak) [inline .travis.yml]: I think this is for JRuby 1.x, which maybe we still support, same for RBX..
Neither of those are failing, so I don't really care too much.

File: Gemfile (hunk 3 above)
Comment (zzak) [inline Gemfile]: Yeah, this is only used for our tests.

File: Gemfile (hunk 4 above)
Comment (zzak) [inline Gemfile]: When I [tried a more elaborate versioning scheme](https://github.com/sinatra/sinatra/commit/a2eecf858a2ad4a844a57baf2b2d187c526ebb00) for versions 2.0 and below [didn't work](https://travis-ci.org/sinatra/sinatra/builds/192352440) and installed the wrong version of nokogiri.

Ignore the errors in that build about i18n, this was fixed by just pinning activesupport and i18n to a much older version that was still supported by everything. Although, admittedly this is really just a consequence of rabl having an open-ended dependency on AS.

**Comment by zzak** [comment]
cc #1231

**Pull request merged**